### ISSUE TITLE
RUF100 fails to catch `# ruff: noqa`

### ISSUE THREAD

**Issue by ddeepwell** [astral-sh/ruff#17042]
### Summary

Rule [RUF100 (unused no-qa)](https://docs.astral.sh/ruff/rules/unused-noqa/) does not catch unused directives when specified as `# ruff: noqa` or `# ruff: noqa: {code}`.

See [playground](https://play.ruff.rs/29158e07-d466-4b38-ae33-5f4e04a90448) for a simplification of the working example of the following three files:

```toml
# ruff.toml
[lint]
select = [ "RUF" ]
```

```python
# example_missed.py
# ruff: noqa
```

```python
# example_caught.py
# noqa
```

Execute command: `ruff check`

### Version

ruff 0.11.2 (4773878ee 2025-03-21)
